src/test/java/org/prebid/server/bidder/gamma/GammaBidderTest.java

```diff
@@ -0,0 +1,328 @@
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<Void>>> result = gammaBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("ext.bidder.publisher not provided");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenNotBannerOrVideoIsPresent() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<Void>>> result = gammaBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(2);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Gamma only supports banner and video media" +
+                " types. Ignoring imp id=");
+        assertThat(result.getErrors().get(1).getMessage()).startsWith("No valid impressions");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenExtImpGammaIdIsBlank() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .banner(Banner.builder().build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpGamma.of("", "zid", "wid"))))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<Void>>> result = gammaBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("PartnerID is empty");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenExtImpGammaWidIsBlank() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .banner(Banner.builder().build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpGamma.of("id", "zid", ""))))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<Void>>> result = gammaBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("WebID is empty");
```

Comment:
We mostly use this style assert for `.startsWith`, when we cannot for sure know the whole message, but know at least first (static) part of it, like you have down in `makeBidsShouldReturnErrorWhenResponseBodyCouldNotBeParsed`. For a simple cases like this, please use a fluent `.containsOnly` with the first assert (where you check size of errors) and compare with expected error with message.

Please check all other cases as well.